File: languageExtensions/OrchidBible/src/main/java/com/eden/orchid/languages/bible/BibleModule.java
package com.eden.orchid.languages.bible;

import com.eden.americanbiblesociety.ABSRepository;
import com.eden.flexmark.BibleVerseExtension;
import com.eden.orchid.api.registration.OrchidModule;
import com.vladsch.flexmark.Extension;
import com.vladsch.flexmark.util.options.MutableDataSet;

public class BibleModule extends OrchidModule {

    @Override
    protected void configure() {
        // Bible verses in Markdown
        addToSet(Extension.class, BibleVerseExtension.create());
        MutableDataSet absOptions = new MutableDataSet();
        absOptions.set(BibleVerseExtension.BIBLE_REPOSITORY, ABSRepository.class);
        addToSet(MutableDataSet.class, absOptions);
    }
}

File: languageExtensions/OrchidBible/src/main/kotlin/com/eden/orchid/languages/bible/BibleModule.kt
package com.eden.orchid.languages.bible

import com.eden.orchid.api.registration.OrchidModule
import com.eden.orchid.api.theme.components.OrchidComponent
import com.eden.orchid.languages.bible.components.ReftaggerComponent
import com.eden.orchid.utilities.addToSet

class BibleModule : OrchidModule() {

    override fun configure() {
        withResources()

        addToSet<OrchidComponent, ReftaggerComponent>()
    }

}
